docs/content/manual/pre-release/resiliency/timeout.md

```diff
@@ -0,0 +1,65 @@
+---
+title: "Test timeout on loss of network connectivity"
+---
+
+## Ping Timeout
+
+1. Create a docker network:
+
+```shell
+docker network create -d bridge --subnet 192.168.22.0/24 longhorn-network
+```
+
+2. Start a replica:
+
+```shell
+docker run --net longhorn-network --ip 192.168.22.2 \
+         -v /volume longhornio/longhorn-engine:<tag> \
+         longhorn replica --listen 192.168.22.2:9502 --size 10g /volume
+```
+
+3. Start another replica:
+
+```shell
+docker run --net longhorn-network --ip 192.168.22.3 \
+         -v /volume longhornio/longhorn-engine:<tag> \
+         longhorn replica --listen 192.168.22.3:9502 --size 10g /volume
+```
+
+4. In another terminal, start the controller:
+
+```shell
+docker run --net longhorn-network --ip 192.168.22.4 --privileged \
+         -v /dev:/dev -v /proc:/host/proc \
+         longhornio/longhorn-engine:<tag> \
+         longhorn controller --replica tcp://192.168.22.2:9502 \
+         --replica tcp://192.168.22.3:9502 \
+         --frontend tgt-blockdev timeout-test
+```
+
+5. In another terminal, find the name of the container running the replica with `docker ps`
+
+6. Disconnect the replica container from the network:
+
+```shell
+docker network disconnect longhorn-network <container id/name>
+```
+
+7. Note that the controller output displays "Backend tcp://192.168.22.2:9502 monitoring failed, mark as ERR: ping timeout" after eight seconds. 
+
+## R/W Timeout
+
+1. Perform steps 2-4 from the Ping Timeout Section.
+
+2. In another terminal, perform I/O on the block device:
+
+```shell
+fio --name=random-writers --ioengine=libaio --rw=randwrite --bs=16k \
+           --direct=1 --size=1000m --numjobs=1 \
+           --filename=/dev/longhorn/timeout-test --iodepth=1
+```
+3. Stop the network on the replica with steps 5-6 in the Ping Timeout Section.
+
+4. Note that the controller output displays ""Setting replica tcp://192.168.22.2:9502 to ERR due to: r/w timeout" after eight seconds.
+
+5. Note that the fio operation continues to work and completes without error.
```

Comment:
Need to check `dmesg` to ensure there is no error from the kernel or filesystem level.

Reply:
I added a step to check dmesg.  This test doesn't use a filesystem on the block device.

Reply:
Hmm? Should we add filesystem then run top of it? That's how normal user use it. And it should be able to expose more problem e.g. filesystem might be more sensitivity.

Reply:
When I used fio on a file on a file system and the replica was getting timed out with old code, the journal of the last 1000 operations showed mostly the operations with the block size of the fio with periodic write operations of 768k and 1280k.  Also, running mkfs.ext4 seemed to use the largest possible block sizes; it was similar to running fio with a blocksize of 1280k.  

It will valuable to use a filesystem because will perform different sized I/O operations and it is what the end user will actually be using.  I'll add a test to create an ext4 filesystem, mount it, and use fio on it.

Reply:
We need filesystem level check on timeout as well, in addition to block level check.

Reply:
Added filesystem check.